primitives/application-crypto/src/lib.rs

```diff
@@ -312,11 +314,11 @@ macro_rules! app_crypto_public_common {
 	};
 }
 
-/// Implements traits for the public key type if `feature = "std"` is enabled.
-#[cfg(feature = "std")]
+/// Implements traits for the public key type if `feature = "serde"` is enabled.
+#[cfg(feature = "serde")]
 #[doc(hidden)]
 #[macro_export]
-macro_rules! app_crypto_public_common_if_std {
+macro_rules! app_crypto_public_common_if_serde {
 	() => {
 		impl $crate::Derive for Public {
```

Comment:
Related to: https://github.com/paritytech/substrate/pull/13027#discussion_r1183342629

Do we need `Derive` in `no-std`?

Reply:
The macro itself is needed in `serde` because it contains serialize and deserialize. Would you like to split the macro?

Regarding if `Derive` should be part of `full_crypto`: Until now it has not been part of it. So I'm not sure if that should be part of this PR or rather in a different issue?

Reply:
Forget what I wrote here 🙈. We need Derive in nostd because we must be able to (de)serialise keys.